pkgs/tools/text/gawk/default.nix

```diff
@@ -74,9 +45,66 @@ stdenv.mkDerivation rec {
 
     license = licenses.gpl3Plus;
 
-    platforms = platforms.unix;
+    platforms = platforms.unix ++ platforms.windows;
 
     maintainers = [ ];
   };
-}
+in
+
+if stdenv.hostPlatform.isWindows then
```

Comment:
Splitting this into two conditions, does not seem maintainable to me in the longterm. Something like:

```
  buildFlags = optionalString stdenv.hostPlatform.isWindows "mingw32";
```

Is totally fine but this seems like too much differentation IMO. We should try to work with gawk upstream to allow using autotools with Mingw builds.

Reply:
This has been done after your [change in zlib](https://github.com/NixOS/nixpkgs/commit/845a364564b2829b3813d5887b85cf98049f4f72) in order to avoid mass-rebuild (and thus going through ```staging```). Unfortunately ```//```-overriding does not work here.

Maintanability of a derivation which has every parameter  (except ```name``` and ```meta```) conditional on ```stdenv.hostPlatform.isWindows``` is questinable as well. And any change would result on mass-rebuild on all platforms.

Reply:
Also, I am not sure if GNU utils are going to stay on nixpkgs/Windows as it is.
We need them to bootstrap ```nix.exe``` but then they can be replaced with something else, more Windows-friendly (```msys2``` or there is a project to rewrite GNU utils in Rust)

Reply:
For example, ```gnutar```, ```xz``` and ```unzip``` might be replaced with ```7z``` (which is able to unpack almose all the archives) instead of trying to fix old GNU/Cygwin code

Reply:
I think we want to stick to GNU stuff as much as possible. They are almost always extremely portable and will gladly accept patches when things don't work.

I am thinking this kind of big change should be left to a fork for now. If we can get Nix to work on Windows then maybe it would be worth doing extra effort to get this stuff to work on Windows. But for now there's lots of issues in running Nix on native Windows that should have a priority over these kind of changes.

Reply:
The problem with GNU code is more not with the code itself, but with ```./configure``` scripts which expect POSIX or an emulation layer. It seems that ```gawk``` addresses excatly this problem by providing results of ```./configure``` in the tarball.
So far we can run ```autotools``` doing cross-compilation, but switching to native compilation would require setting up Cygwin/WSL/MSYS only to run ```autotools```, while ```cmake```/```cargo```/```ninja```/..-based project would not require it. And ```gawk``` with its providing results of ```./configure``` too.